web-common/src/features/dashboards/state-managers/loaders/explore-active-page-store.ts

```diff
@@ -0,0 +1,310 @@
+}
+
+export function updateExploreSessionStore(
+  exploreName: string,
+  storageNamespacePrefix: string | undefined,
+  metricsView: V1MetricsViewSpec,
+  explore: V1ExploreSpec,
+  timeControlsState: TimeControlState | undefined,
+  exploreState: MetricsExplorerEntity,
+) {
+  let activePage = exploreState.activePage;
+  if (activePage === DashboardState_ActivePage.DIMENSION_TABLE) {
+    activePage = DashboardState_ActivePage.DEFAULT;
+  }
+
+  const storedExploreState: Partial<MetricsExplorerEntity> = {};
+  const sharedExploreState: Partial<MetricsExplorerEntity> = {
+    ...exploreState,
+  };
+
+  for (const key of ExploreActivePageKeys[activePage]) {
+    storedExploreState[key] = exploreState[key] as any;
+    delete sharedExploreState[key];
+  }
+  for (const key of ExploreActivePageOtherKeys[activePage]) {
+    storedExploreState[key] = exploreState[key] as any;
+    delete sharedExploreState[key];
+  }
+  storedExploreState.activePage = Number(storedExploreState.activePage);
+
+  const storedUrlSearch = convertExploreStateToURLSearchParams(
+    storedExploreState as MetricsExplorerEntity,
+    explore,
+    timeControlsState,
+    {},
+  );
+  const sharedUrlSearch = convertExploreStateToURLSearchParams(
+    sharedExploreState as MetricsExplorerEntity,
+    explore,
+    timeControlsState,
+    {},
+  );
+  try {
+    setExploreStateForActivePage(
+      exploreName,
+      storageNamespacePrefix,
+      activePage,
+      storedUrlSearch.toString(),
+    );
+    setSharedExploreState(
+      exploreName,
+      storageNamespacePrefix,
+      sharedUrlSearch.toString(),
+    );
+  } catch {
+    // no-op
+  }
+
+  for (const otherPage in ExploreActivePageSharedKeys[activePage]) {
+    const sharedKeys = ExploreActivePageSharedKeys[activePage][otherPage];
+    if (!sharedKeys?.length) continue;
+
+    const otherPageKey = getKeyForSessionStore(
+      exploreName,
+      storageNamespacePrefix,
+      Number(otherPage),
+    );
+    const otherPageRawPreset = sessionStorage.getItem(otherPageKey) ?? "";
+
+    const { partialExploreState: otherPageExploreState } =
+      convertURLSearchParamsToExploreState(
+        new URLSearchParams(otherPageRawPreset),
+        metricsView,
+        explore,
+        {},
+      );
+
+    for (const sharedKey of sharedKeys) {
+      if (!(sharedKey in storedExploreState)) continue;
+      otherPageExploreState[sharedKey] = storedExploreState[sharedKey];
+    }
+    const otherPageUrlSearch = convertExploreStateToURLSearchParams(
+      otherPageExploreState as MetricsExplorerEntity,
+      explore,
+      timeControlsState,
+      {},
+    );
+    try {
+      setExploreStateForActivePage(
+        exploreName,
+        storageNamespacePrefix,
+        Number(otherPage),
+        otherPageUrlSearch.toString(),
+      );
+    } catch {
+      // no-op
+    }
+  }
+}
+
+export function getExplorePresetForActivePage(
```

Comment:
This should return an `ExploreState` (`MetricsExplorerEntity`) not an `ExplorePreset`